src/Servers/Kestrel/Core/src/Internal/Http/HttpRequestHeaders.cs

```diff
@@ -11,8 +11,14 @@
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http
 {
-    public partial class HttpRequestHeaders : HttpHeaders
+    public sealed partial class HttpRequestHeaders : HttpHeaders
     {
+        public void Reset()
+        {
+            _isReadOnly = false;
+            ClearFast();
```

Comment:
It doesn't feel like hoisting the Reset() method to the derived sealed class should be necessary for devirtualization. Is there a reason this is necessary that I'm not aware of? Or maybe a CoreCLR that track devirtualizing in this case?

Reply:
Can't find the comment about it; but for the method in the parent class the type would be `HttpHeaders` as its a shared method between all derived types so the Jit can't "see" which single implementation of the override of `ClearFast` to call (as it could be many based on the actual type).

With the method in the sealed derived type there is only one override of `ClearFast` to call so the Jit can "see" that and devirtualize it.

/cc @AndyAyersMS

Reply:
OTOH if `Reset()` inlined into the caller then it could devirtualize there as the dervied types are explicit, so might get the same effect from just sealing (and adding AggressiveInlining to be sure)

Reply:
Changed to the inline approach